akka-stream/src/main/scala/akka/stream/impl/io/ByteStringParser.scala

```diff
@@ -78,6 +111,8 @@ private[akka] abstract class ByteStringParser[T] extends GraphStage[FlowShape[By
  */
 private[akka] object ByteStringParser {
 
+  val CompactionThreshold = 16
```

Comment:
Contacted my astrology advisor :D I will be honest, I don't know what is the ideal number here. It should be a finite number and not too small to reduce the copying.